tests/mir-opt/sroa/lifetimes.rs

```diff
@@ -16,6 +15,9 @@ struct Foo<T: Err> {
 
 // EMIT_MIR lifetimes.foo.ScalarReplacementOfAggregates.diff
 fn foo<T: Err>() {
+    // CHECK-LABEL: fn foo
```

Comment:
As said in the comment, this test is about verifying that we erase the lifetimes in the types of locals. Could you add the annotations for this?